wandering-warriors/main.py

```diff
@@ -27,9 +33,71 @@ class Ledger(Widget):
 class OperationsBar(Widget):
     pass
 
-
-class CuneiformDrawingInput(Widget):
-    pass
+def calculate_points(x1, y1, x2, y2, steps=1):
+    dx = x2 - x1
+    dy = y2 - y1
+    dist = sqrt(dx * dx + dy * dy)
+    if dist < steps:
+        return
+    o = []
+    m = dist / steps
+    for i in range(1, int(m)):
+        mi = i / m
+        lastx = x1 + dx * mi
+        lasty = y1 + dy * mi
+        o.extend([lastx, lasty])
+    return o
+
+class CuneiformDrawingInput(FloatLayout):
+
+    def on_touch_down(self, touch):
+        ud = touch.ud
+        if(touch.pos[0] > self.pos[0] and 
+           touch.pos[0] < self.pos[0] + self.size[0] and 
+           touch.pos[1] > self.pos[1] and 
+           touch.pos[1] < self.pos[1] + self.size[1]):
+                ud['group'] = g = str(touch.uid)
+                pointsize = 1
+                ud['color'] = 0
+                with self.canvas:
+                    Color(0, 0, 0)
+                    ud['lines'] = [
+                        Point(points=(touch.x, touch.y), source='particle.png',
+                              pointsize=pointsize, group=g)]
+                return True
+        else:
+            ud['lines'] = []
+
+    def on_touch_move(self, touch):
```

Comment:
Does this cover `on_touch_down` too?

Reply:
Hmm no that's probably why I resorted to the try catch.

Reply:
Actually the change to ud does carry over because it changes touch.ud, I Believe